File: Estilos.Arquitectura/Arquitectura.Cliente.Servidor/UserService.cs
namespace Arquitectura.Cliente.Servidor
{
    public class UserService
    {
        private readonly UserRepository _userRepository;

        public UserService()
        {
            _userRepository = new UserRepository();
        }

        public User GetUserById(string id)
        {
            return _userRepository.GetUserById(id);
        }

        public void SaveUser(User user)
        {
            _userRepository.SaveUser(user);
        }
    }
}

File: Estilos.Arquitectura/Arquitectura-capas/src/main/java/org/example/service/UserService.java
package org.example.service;

import org.example.model.User;
import org.example.repository.UserRepository;

import java.util.List;

public class UserService {
    private UserRepository userRepository;

    public UserService() {
        this.userRepository = new UserRepository();
    }

    public User getUserById(String id) {
        return userRepository.getUserById(id);
    }

    public void saveUser(User user) {
        userRepository.saveUser(user);
    }

    public void updateUser(User user) {
        userRepository.updateUser(user);
    }

    public void deleteUser(String id) {
        userRepository.deleteUser(id);
    }

    public List<User> getAllUsers() {
        return userRepository.getAllUsers();
    }
}
